test/suites/examples/install-karpenter-kit.yaml

```diff
@@ -0,0 +1,114 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: install-karpenter-kit-task
+  namespace: karpenter-tests
+spec:
+  description: |
+    Install Karpenter and necessary resources onto an EKS cluster.
+  params:
+  - name: test-cluster-name
+    description: The name of the EKS cluster to install Karpenter
+  - name: karpenter-version
+    description: The version of Karpenter to install. Needs to be in vx.y.z format where x, y, and z are integers.
+  workspaces:
+  - name: config
+    description: |
+      Directory where kubeconfig for the Guest Cluster lives
+  steps:
+  - name: karpenter-cfn-resources
+    image: public.ecr.aws/karpenter-testing/minimal:latest
+    script: |
+      echo "Getting Guest Cluster kube-admin config"
+      kubectl get secret -n karpenter-tests $(params.test-cluster-name)-kube-admin-config -ojsonpath='{.data.config}' | base64 -d > $(workspaces.config.path)/kubeconfig
+
+      TEMPOUT=$(mktemp)
+      curl -fsSL "https://karpenter.sh/"$(params.karpenter-version)"/getting-started/getting-started-with-eksctl/cloudformation.yaml"  > $TEMPOUT \
+      && aws cloudformation deploy \
+        --stack-name "KarpenterTesting-$(params.test-cluster-name)" \
+        --template-file "${TEMPOUT}" \
+        --capabilities CAPABILITY_NAMED_IAM \
+        --parameter-overrides "ClusterName=$(params.test-cluster-name)"
+
+      export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --output text --query Account --no-cli-pager)
+      aws iam attach-role-policy --role-name KitDPRole-$(params.test-cluster-name) --policy-arn arn:aws:iam::${AWS_ACCOUNT_ID}:policy/KarpenterControllerPolicy-$(params.test-cluster-name)
+
+  - name: spot-slr
+    image: amazon/aws-cli
+    script: |
+      aws iam create-service-linked-role --aws-service-name spot.amazonaws.com || true
+
+  - name: helm-install-karpenter
+    image: alpine/helm:latest
+    script: |
+      export CLUSTER_ENDPOINT="$(cat $(workspaces.config.path)/kubeconfig | grep server | awk '{print $2}')"
+
+      helm repo add karpenter https://charts.karpenter.sh/
+      helm repo update
+
+      helm upgrade --kubeconfig=$(workspaces.config.path)/kubeconfig --install --namespace karpenter --create-namespace \
+        karpenter karpenter/karpenter \
+        --version $(params.karpenter-version) \
+        --set clusterName=$(params.test-cluster-name) \
+        --set clusterEndpoint=${CLUSTER_ENDPOINT} \
+        --set aws.defaultInstanceProfile=KarpenterNodeInstanceProfile-$(params.test-cluster-name)
+
+  - name: create-provisioner
+    image: public.ecr.aws/karpenter-testing/minimal:latest
+    script: |
+      cat <<EOF | kubectl --kubeconfig=$(workspaces.config.path)/kubeconfig apply -f -
+      apiVersion: karpenter.sh/v1alpha5
+      kind: Provisioner
+      metadata:
+        name: default
+      spec:
+        limits:
+          resources:
+            cpu: 1k
+        provider:
+          instanceProfile: KitDP-$(params.test-cluster-name)
```

Comment:
We have the option to not have this step in the install, but would need this instanceProfile for this to work.